tox.ini

```diff
@@ -4,9 +4,72 @@ envlist = {dev,rhcs}-{centos,ubuntu}-{container,non_container}-{all_daemons,coll
   {dev,rhcs}-{centos,ubuntu}-non_container-{switch_to_containers}
   dev-rhel-container-podman
   infra_lv_create
+  migrate_ceph_disk_to_ceph_volume
 
 skipsdist = True
 
+# a test scenario for the migrate-ceph-disk-to-ceph-volume.yml playbook
+[testenv:migrate_ceph_disk_to_ceph_volume]
+whitelist_externals =
+    vagrant
+    bash
+    cp
+    git
+    pip
+passenv=*
+setenv=
+  ANSIBLE_SSH_ARGS = -F {changedir}/vagrant_ssh_config
+  ANSIBLE_CONFIG = -F {toxinidir}/ansible.cfg
+  ANSIBLE_ACTION_PLUGINS = {toxinidir}/plugins/actions
+  ANSIBLE_CALLBACK_PLUGINS = {toxinidir}/plugins/callback
+  ANSIBLE_CALLBACK_WHITELIST = profile_tasks
+  CEPH_STABLE_RELEASE = luminous
+  # only available for ansible >= 2.5
+  ANSIBLE_STDOUT_CALLBACK = yaml
+changedir={toxinidir}/tests/functional/migrate_ceph_disk_to_ceph_volume
+commands=
+  vagrant up --no-provision {posargs:--provider=virtualbox}
+  bash {toxinidir}/tests/scripts/generate_ssh_config.sh {changedir}
+
+  # use the stable-3.2 branch to deploy a luminous cluster
+  git clone -b {env:CEPH_ANSIBLE_BRANCH:stable-3.2} --single-branch https://github.com/ceph/ceph-ansible.git {envdir}/tmp/ceph-ansible
+  pip install -r {envdir}/tmp/ceph-ansible/tests/requirements.txt
+
+  ansible-playbook -vv -i {changedir}/hosts {envdir}/tmp/ceph-ansible/tests/functional/setup.yml
+
+  # deploy the cluster
+  ansible-playbook -vv -i {changedir}/hosts {envdir}/tmp/ceph-ansible/{env:PLAYBOOK:site.yml.sample} --extra-vars "\
+      delegate_facts_host={env:DELEGATE_FACTS_HOST:True} \
+      fetch_directory={env:FETCH_DIRECTORY:{changedir}/fetch} \
+      ceph_stable_release={env:CEPH_STABLE_RELEASE:luminous} \
+      ceph_docker_registry={env:CEPH_DOCKER_REGISTRY:docker.io} \
+      ceph_docker_image={env:CEPH_DOCKER_IMAGE:ceph/daemon} \
+      ceph_docker_image_tag={env:CEPH_DOCKER_IMAGE_TAG:latest-master} \
+      ceph_dev_branch={env:CEPH_DEV_BRANCH:master} \
+      ceph_dev_sha1={env:CEPH_DEV_SHA1:latest} \
+      copy_admin_key={env:COPY_ADMIN_KEY:False} \
+  "
+
+  # wait 30sec for services to be ready
+  sleep 30
+  # test cluster state using ceph-ansible tests
+  py.test -n 8 --durations=0 --sudo -v --connection=ansible --ansible-inventory={changedir}/hosts {envdir}/tmp/ceph-ansible/tests/functional/tests
+
+  # migrate osds to ceph-volume
+  cp {toxinidir}/infrastructure-playbooks/migrate-ceph-disk-osds-to-ceph-volume.yml {toxinidir}/migrate-ceph-disk-osds-to-ceph-volume.yml
+  ansible-playbook -vv -i {changedir}/hosts {toxinidir}/migrate-ceph-disk-osds-to-ceph-volume.yml --extra-vars "ireallymeanit=yes"
```

Comment:
It should be run using ansible 2.7 while it runs against ansible 2.6 because of the command : 

`pip install -r {envdir}/tmp/ceph-ansible/tests/requirements.txt` executed earlier after the git clone (L36).

```
ansible-playbook 2.6.14
  config file = None
  configured module search path = [u'/home/jenkins-build/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /tmp/tox.3uD2995ofX/migrate_ceph_disk_to_ceph_volume/lib/python2.7/site-packages/ansible
  executable location = /tmp/tox.3uD2995ofX/migrate_ceph_disk_to_ceph_volume/bin/ansible-playbook
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
No config file found; using defaults

PLAYBOOK: migrate-ceph-disk-osds-to-ceph-volume.yml ****************************
1 plays in /home/jenkins-build/build/workspace/ceph-ansible-scenario/migrate-ceph-disk-osds-to-ceph-volume.yml
```

I think we need to reinstall dependencies from the main repo by executing :
`pip install -r {toxinidir}/tests/requirements.txt` before running `migrate-ceph-disk-osds-to-ceph-volume.yml`

Reply:
I was conflicted about this. If you think it's best to run ``migrate-ceph-disk-osds-to-ceph-volume.yml`` under Ansible 2.7 then I'll change it but my thought was that this playbook would be run against luminous or mimic clusters that probably have Ansible 2.6 installed.

Reply:
Talking in IRC it was decided to test the new playbook with Ansible 2.7. I started another test here: https://2.jenkins.ceph.com/job/ceph-ansible-scenario/534/console